File: cpp/896_单调序列_easy.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        int inc = 0, des = 0;
        int N = A.size();
        for (int i=1; i<N; i++) {
            if (A[i] > A[i-1] && inc == 0)
                inc++;
            else if (A[i] < A[i-1] && des == 0)
                des++;
        }
        return inc & des ? false : true;
    }
};

File: python/896_单调序列_easy.py
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        inc, des = 0, 0
        N = len(A)
        for i in range(1, N):
            if A[i] > A[i-1] and inc == 0:
                inc += 1
            elif A[i] < A[i-1] and des == 0:
                des += 1
        return False if inc & des else True